app/jobs/cache/browser_channels/main.rb

```diff
@@ -0,0 +1,21 @@
+class Cache::BrowserChannels::Main
+  include Sidekiq::Worker
+  sidekiq_options queue: :scheduler, retry: false
+
+  LAST_RAN_AT_KEY = "cache_browser_channels_main_last_ran_at".freeze
+  RESPONSES_PREFIX_LENGTH = 3
+  def perform
+    previous_run = Rails.cache.fetch(LAST_RAN_AT_KEY)
+    return if previous_run.present? && previous_run.to_time >= 2.hours.ago
+    Cache::BrowserChannels::PrefixList.perform_async
+    sql = "SELECT DISTINCT SUBSTRING(sha2_base16, 1, #{RESPONSES_PREFIX_LENGTH}) as prefix FROM site_banner_lookups"
+    sql += " WHERE wallet_status != 0"
+    sql += " AND updated_at >= '#{(previous_run.to_time - 3.minutes).to_s}'" if previous_run.present?
+    sql += " ORDER BY prefix desc"
+    result = ActiveRecord::Base.connection.execute(sql)
```

Comment:
One thing that I noticed while [reading the docs](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-execute) on this function call is memory might need to be cleared after this call.

>Note: depending on your database connector, the result returned by this method may be manually memory managed. Consider using the exec_query wrapper instead.

It's fine to keep this as is but a nit I have is that this isn't using SQL parameters. 

I think I'd prefer to see something like this
```ruby
if previous_run.present?
  previous_run.to_time - 3.minutes)
else 
  previous_run = Time.at(0)
end

ActiveRecord::Base.connection.execute("
  SELECT DISTINCT SUBSTRING(sha2_base16, 1, :response_prefix_length) as prefix FROM site_banner_lookups
  WHERE wallet_status != 0
  AND updated_at >= :updated_at
  ORDER BY prefix desc",
  response_prefixe_length: RESPONSES_PREFIX_LENGTH,
  updated_at: previous_run
)
```

Reply:
I checked the documentation, since we get a `PG::Result`, the documentation says it'll get added to Ruby's garbage collector: https://www.rubydoc.info/gems/pg/PG/Result 

My impression is that Ruby uses the mark & sweep garbage collection algorithm, so I think we'll be okay.